atom/common/node_bindings.cc

```diff
@@ -213,19 +215,69 @@ void NodeBindings::Initialize() {
   // Explicitly register electron's builtin modules.
   RegisterBuiltinModules();
 
-  // Init node.
-  // (we assume node::Init would not modify the parameters under embedded mode).
-  // NOTE: If you change this line, please ping @codebytere or @MarshallOfSound
-  int argc = 0;
+  // pass non-null program name to argv so it doesn't crash
+  // trying to index into a nullptr
+  int argc = 1;
   int exec_argc = 0;
-  const char** argv = nullptr;
+  const char* prog_name = "electron";
+  const char** argv = &prog_name;
   const char** exec_argv = nullptr;
+
+  std::unique_ptr<base::Environment> env(base::Environment::Create());
+  if (env->HasVar("NODE_OPTIONS")) {
+    bool is_packaged_app = false;
+
+    base::FilePath exe_path;
+    base::PathService::Get(base::FILE_EXE, &exe_path);
+    std::string path = exe_path.value();
+    std::transform(path.begin(), path.end(), path.begin(), ::tolower);
+
+#if defined(OS_WIN)
+    is_packaged_app = path == "electron.exe";
```

Comment:
_(minor)_ the advance declaration above could be removed and these two lines could be

```cc
#if defined(OS_WIN)
  const bool is_packaged_app = path == "electron.exe";
#else
  const bool is_packaged_app = path == "electron";
#endif